src/main/scala/cromwell/engine/workflow/CallMetadataBuilder.scala

```diff
@@ -0,0 +1,190 @@
+package cromwell.engine.workflow
+
+import cromwell.binding._
+import cromwell.engine.backend.{CallMetadata, StdoutStderr}
+import cromwell.engine.db.ExecutionDatabaseKey
+import cromwell.engine.db.slick._
+import cromwell.engine.ExecutionIndex._
+
+import scala.language.postfixOps
+
+/**
+ * Builds call metadata suitable for return as part of a workflow metadata request.
+ */
+object CallMetadataBuilder {
+
+  // Collects up all the data for a single CallMetadata in the easiest-to-assemble format, but
```

Comment:
I get what this comment is saying but think it's a little confusing. Can you explain a bit more, e.g. Why easiest-to-assemble != required for endpoint